src/Components/WebAssembly/WebAssembly.Authentication/src/Services/RemoteAuthenticationService.cs

```diff
@@ -211,6 +233,48 @@ private void UpdateUser(Task<ClaimsPrincipal> task)
 
         static async Task<AuthenticationState> UpdateAuthenticationState(Task<ClaimsPrincipal> futureUser) => new AuthenticationState(await futureUser);
     }
+
+    /// <inheritdoc/>
+    protected virtual void Dispose(bool disposing)
+    {
+        if (!disposedValue)
+        {
+            if (disposing)
+            {
+                _loggerObjectRef?.Dispose();
+            }
+
+            disposedValue = true;
+        }
+    }
+
+    void IDisposable.Dispose()
+    {
+        Dispose(disposing: true);
+        GC.SuppressFinalize(this);
```

Comment:
Do we really need to do all this dispose pattern in full? I don't think the resource we're protecting here (`_loggerObjectRef`) actually benefits, because:

1. `RemoteAuthenticationService` can't be GCed if it's still rooted
2. If `RemoteAuthenticationService` is still rooted, then `_loggerObjectRef` is also still rooted
3. If `_loggerObjectRef` is still rooted, then telling JS to discard it makes no difference

I suspect this full Dispose pattern might be relevant in some edge cases for a JSObjectReference, but not for a DotNetObjectReference. If I've miscalculated, please correct me :)

Reply:
I `ctrl+.` my way to victory, technically you can extend `RemoteAuthenticationService` to tweak the auth operations, hence why the full dispose pattern here. It is what you would do if you wanted to plug in a complete separate implementation of the auth operations.

Reply:
I have removed this.